pkg/s3store/s3store.go

```diff
@@ -1249,3 +1335,95 @@ func (store S3Store) releaseUploadSemaphore() {
 	store.uploadSemaphore.Release()
 	store.uploadSemaphoreDemandMetric.Dec()
 }
+
+func (store S3Store) copyToResponseWriter(w http.ResponseWriter, r io.Reader) (int64, error) {
```

Comment:
Can you explain what the benefit of this is over `io.Copy`?

Reply:
TBH, I rely on AI for development and a lot of this I was able to complete due to its effort. The data copy and range logic is the most complicated parts, and as I did not have a good reason to remove them based on what I saw, I just ensured it was all valid.

If needed, I can replace with io.Copy and move on.